mbq/client/contrib/permissions.py

```diff
@@ -0,0 +1,149 @@
+import json
+import logging
+from typing import Dict, List, Optional, Union
+from uuid import UUID
+
+from typing_extensions import Literal, Protocol
+
+from .. import ServiceClient
+
+logger = logging.getLogger(__name__)
+
+UUIDType = Union[str, UUID]
+# External type returned from internal and external OS Core clients. Keys
+# are the org refs if UUIDs, or {ref_type}:{ref_id} if legacy int types.
+# Values are lists of scopes. Should also include a "global" literal key.
+FetchedPermissionsDoc = Dict[str, List[str]]
+# Internal type stored in the cache. Keys are cache keys with prefixes, colons, etc.
+# Values are pipe-delimited strings with an additional pipe on the end.
+CachedPermissionsDoc = Dict[str, str]
+RefType = Union[Literal["company", "vendor"]]
+
+
+class OSCoreClient(Protocol):
+    def fetch_permissions(
+        self, person_id: str, org_ref: str, ref_type: Optional[RefType] = None
+    ) -> FetchedPermissionsDoc:
+        ...
+
+
+class OSCoreServiceClient:
+    def __init__(self, client: ServiceClient):
+        self.client = client
+
+    def fetch_permissions(
+        self, person_id: str, org_ref: str, ref_type: Optional[RefType] = None
+    ) -> FetchedPermissionsDoc:
+        logger.info(f"Fetching permissions from OS Core: {person_id}, {org_ref}, {ref_type}")
+
+        if ref_type:
+            doc = self.client.get(f"/api/v1/people/{person_id}/permissions/{ref_type}/{org_ref}")
+        else:
+            doc = self.client.get(
+                f"/api/v1/people/{person_id}/permissions", params={"org_ref": org_ref}
+            )
+
+        # Temporary
+        doc["global"] = []
```

Comment:
Are you doing this because consumers of this function expect the global key to be in the dictionary?

Reply:
Just making it work for now, I need the endpoints to provide this (with data in it) eventually